### PR TITLE
[MESH] scalar color settings depending on classification

### PR DESCRIPTION
**Pull request by vcloarec**
In this PR, MDAL reads classified scalar values in dataset. When values are classified, the default scalar rendering settings has color ramp shader and labels according to the classified values.

image

### CODE REVIEW

File: src/app/mesh/qgsmeshrendereractivedatasetwidget.cpp [hunk 1]
```diff
@@ -185,6 +185,11 @@ QString QgsMeshRendererActiveDatasetWidget::metadata( QgsMeshDatasetIndex datase
   const auto options = gmeta.extraOptions();
   for ( auto it = options.constBegin(); it != options.constEnd(); ++it )
   {
+    if ( it.key() == QStringLiteral( "classification" ) )
+    {
+      msg += QStringLiteral( "<tr><td>Classified values</td></tr>" );
```
Comment (nyalldawson) [inline src/app/mesh/qgsmeshrendereractivedatasetwidget.cpp:190]: Translate

**Comment by nyalldawson** [comment]
I'm not so familiar with the dataset capabilities - to clarify, do some mesh formats support embedded classification metadata?

**Comment by vcloarec** [comment]
@nyalldawson some formats can provide values that can be compressed by categorizing values in consecutive intervals, each represent by an integer. Now MDAL has the capabilities to recognize this dataset type and store the bounds of each class an the units in the metadata.
QGIS uses this metadata to setup adapted color ramp shader.

**Comment by nyalldawson** [comment]
Thanks for the clarification @vcloarec !

QGIS code looks good to me, can't comment on the mdal side :)

**Comment by PeterPetrik** [comment]
MDAL already reviewed in the MDAL. repository. all looks good

**Pull request merged**